src/utils/s3_mods.py

```diff
@@ -167,3 +174,341 @@ def rd_write_feather(filepath, df):
+
+    """
+    if filepath is None:
+        response = False
+
+    if rd_file_exists(filepath):
+        isdir = rd_isdir(filepath)
+        size = rd_file_size(filepath)
+        response = (not isdir) and (size > 0)
+    else:
+        response = False
+    return response
+
+
+def rd_stat_size(path: str) -> int:
+    """
+    Gets the file size of a file or directory in bytes.
+    Alias of as rd_file_size.
+    Works for directories, but returns 0 bytes, which is typical for s3.
+    """
+    return rd_file_size(path)
+
+
+def rd_read_header(path: str) -> str:
+    """
+    Reads the first line of a file on s3. Gets the entire file using boto3 get_objects,
+    converts its body into an input stream, reads the first line and remove the carriage
+    return character (backslash-n) from the end.
+
+    Args:
+        filepath (string): The "directory" path in s3 bucket.
+
+    Returns:
+        status (bool): True if the dirpath is a directory, false otherwise.
+    """
+    # Create an input/output stream pointer, same as open
+    stream = TextIOWrapper(s3_client.get_object(Bucket=s3_bucket, Key=path)['Body'])
+
+    # Read the first line from the stream
+    response = stream.readline()
+
+    # Remove the last character (carriage return, or new line)
+    response = response[:-1]
+
+    return response
+
+
+def rd_write_string_to_file(content: bytes, filepath: str):
+    """
+    Writes a string into the specified file path
+    """
+
+    # Put context to a new Input-Output buffer
+    str_buffer = StringIO(content.decode("utf-8"))
+
+    # "Rewind" the stream to the start of the buffer
+    str_buffer.seek(0)
+
+    # Write the buffer into the s3 bucket
+    _ = s3_client.put_object(
+        Bucket=s3_bucket, Body=str_buffer.getvalue(), Key=filepath
+    )
+    return None
+
+
+def _path_long2short(path: "str") -> str:
+    """
+    Extracts a short file name from the full path.
+    If there is at least one forward slash, finds the lates slash to the right
+    and rerurns all characrers afrer it.
+
+    If there are no slashes, returns the path as is.
+    """
+    if "/" in path:
+        last_slash = path.rfind("/")
+        return path[last_slash + 1:]
+    else:
+        return path
+
+
+def _remove_end_slashes(path: "str") -> str:
+    """
+    Removes any amount of consequitive forward slashes from a path.
+    """
+    while path.endswith("/"):
+        path = path[:-1]
+
+    return path
+
+
+def rd_copy_file(src_path: str, dst_path: str) -> bool:
+
+    dst_path = _remove_end_slashes(dst_path)
+    dst_path += "/" + _path_long2short(src_path)
+    """
+    Copy a file from one location to another. Uses rdsa_utils.
+    """
+    success = copy_file(
+        client=s3_client,
+        source_bucket_name=s3_bucket,
+        source_object_name=src_path,
+        destination_bucket_name=s3_bucket,
+        destination_object_name=dst_path,
+    )
+    return success
```

Comment:
I tested this and rd_move_file when the file already exists in the destination. In this case, the return is NOT True.
So for further work, it would be useful to put this in docstrings.